include/coro/net/tcp/server.hpp

```diff
@@ -58,8 +63,19 @@ class server
      * Accepts an incoming tcp client connection.  On failure the tls clients socket will be set to
      * and invalid state, use the socket.is_value() to verify the client was correctly accepted.
      * @return The newly connected tcp client connection.
+     * @note Unix only
      */
-    auto accept() -> coro::net::tcp::client;
+    auto accept() const -> coro::net::tcp::client;
+#endif
+
+    /**
+     * Asynchronously accepts an incoming TCP client connection.
+     * If no connection is received before the internal timeout or cancellation, the result will be std::nullopt.
+     *
+     * @return A task resolving to an optional TCP client connection. The value will be set if a client was
+     *         successfully accepted, or std::nullopt if the operation timed out or was cancelled.
+     */
+    auto accept_client(std::chrono::milliseconds timeout = std::chrono::milliseconds{0}) -> coro::task<std::optional<coro::net::tcp::client>>;
```

Comment:
Can we unify these as possibly:

`auto accept(std::chrono::milliseconds timeout = std::chrono::milliseconds{0}) -> coro::task<std::optional<coro::net::tcp::client>>;`?

If the linux/bsd versions never return `std::nullopt` that is ok, I think its probably more important to have a unified API.